qgis_hub_plugin/gui/resource_browser.py

```diff
@@ -174,30 +179,35 @@ def show_preview(self):
 
     def download_resource(self):
         resource = self.selected_resource()
-        file_path = download_resource_file(
-            resource.file, resource.resource_type, resource.uuid
-        )
+        file_extension = os.path.splitext(resource.file)[1]
+
+        file_path = QFileDialog.getSaveFileName(
+            self, "Save Resource", "", f"All Files (*{file_extension})"
```

Comment:
Set the filename from the `resource.file,` so that the user doesn't need to come up with something new.

Reply:
I guess, the filter should be something like this:
```
"All Files (*);;Geopackage (*.gpkg);;QGIS Model (*.model3),"
```

And again, if you find a string, put it in a `self.tr`